class.jetpack.php

```diff
@@ -1041,6 +1049,103 @@ public function check_twitter_tags() {
 			require_once JETPACK__PLUGIN_DIR . 'class.jetpack-twitter-cards.php';
 		}
 	}
+	
+	
+	
+	
+	/*
+	
+	Example code to submit a security report:
+	
+	function akismet_submit_jetpack_security_report() {
+		Jetpack::submit_security_report( 'spam', 'akismet', $args = array( 'plugin' => 'Akismet', 'blocked' => 138284 ) );
+	}
+	add_action( 'jetpack_security_report', 'akismet_submit_jetpack_security_report' );
+	
+	*/
+	
+	/**
+	 * Allows plugins to self report for our security dashboard.
+	 *
+	 * @return null
+	 */
+	public function perform_security_reporting() {
+		$security_report_transient = get_transient( 'jetpack_security_report' );
+		if( $security_report_transient ) {
+			return;
+		}
+		
+		do_action( 'jetpack_security_report' );
+		
+		Jetpack_Options::update_option( 'security_report', $this->security_report );
+		set_transient( 'jetpack_security_report', 1, 15 * MINUTE_IN_SECONDS );
+	}
+	
+	// types: login_form, backup, file_scanning, spam
+	// args for login_form and spam: 'blocked'=>(int)(optional), 'status'=>(string)(ok, warning, error), 'message'=>(optional, disregarded if status is ok, allowed tags: a, em, strong)
+	// args for backup and file_scanning: 'last'=>(timestamp)(optional), 'next'=>(timestamp)(optional), 'status'=>(string)(ok, warning, error), 'message'=>(optional, disregarded if status is ok, allowed tags: a, em, strong)
+	function submit_security_report( $type = '', $plugin_file = '', $args = array() ) {	
+		if( !is_string( $type ) || !is_string( $plugin_slug ) ) {
+			return new Jetpack_Error( 'invalid_security_report', 'Invalid Security Report' );
+		}
+		
+		//Get rid of any non-allowed args
+		$args = array_intersect_key( $args, array( 'blocked', 'last', 'next', 'status', 'message' ) );
+		
+		$plugin = get_plugin_data( $plugin_file );
+		
+		if ( !$plugin['Name'] ) {
+			return new Jetpack_Error( 'security_report_missing_plugin_name', 'Invalid Plugin File Provided' );
```

Comment:
Because it's within Jetpack and Jetpack_Error is a wrapper for WP_Error, so I figured that was the preferred way to handle